UnityProject/Assets/Scripts/HealthV2/Living/PolymorphicSystems/Bodypart/BodyPartComponentBase.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using System.Linq;
+
+namespace HealthV2.Living.PolymorphicSystems.Bodypart
+{
+	public abstract class BodyPartComponentBase : BodyPartFunctionality
+	{
+		public override void RemovedFromBody(LivingHealthMasterBase livingHealth)
+		{
+			foreach (var sys in livingHealth.ActiveSystems)
+			{
+				sys.BodyPartRemoved(RelatedPart);
+			}
+		}
+
+		public override void AddedToBody(LivingHealthMasterBase livingHealth)
+		{
+			if(HasSystem(livingHealth) == false)
+			{
+				var sys = GenSystem(livingHealth);
+				sys.Base = livingHealth;
+				sys.InIt();
+				livingHealth.ActiveSystems.Add(sys);
+			}
+
+			foreach (var sys in livingHealth.ActiveSystems)
+			{
+				sys.BodyPartAdded(RelatedPart);
+			}
+		} //Warning only add body parts do not remove body parts in this
```

Comment:
I think people can tell that they should only add body parts on the "addedtobody" function